designer/server/src/main/scala/pl/touk/nussknacker/ui/additionalconfig/AdditionalUIConfigProvider.scala

```diff
@@ -0,0 +1,71 @@
+package pl.touk.nussknacker.ui.additionalconfig
+
+import pl.touk.nussknacker.engine.api.component.{
+  ComponentGroupName,
+  ComponentId,
+  ParameterConfig,
+  ScenarioPropertyConfig,
+  SingleComponentConfig
+}
+import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
+import pl.touk.nussknacker.engine.definition.component.parameter.editor.EditorExtractor
+import pl.touk.nussknacker.engine.graph.node.FragmentInputDefinition.{
+  ParameterValueCompileTimeValidation,
+  ValueInputWithFixedValues
+}
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+
+/**
+ * Trait allowing the provision of UI configuration for components and scenario properties, without requiring a model reload.
+ *
+ * TODO: The current implementation allows providing configs only for standard components - meaning that base components aren't handled.
+ */
+trait AdditionalUIConfigProvider extends Serializable {
+
+  def getAllForProcessingType(processingType: String): Map[ComponentId, AdditionalUIConfig]
+
+  def getScenarioPropertiesUIConfigs(processingType: String): Map[String, ScenarioPropertyConfig]
+
+}
+
+object AdditionalUIConfigProvider {
+  val empty = new DefaultAdditionalUIConfigProvider(Map.empty, Map.empty)
+}
+
+case class AdditionalUIConfig(
+    parameterConfigs: Map[String, ParameterAdditionalUIConfig],
+    icon: Option[String] = None,
+    docsUrl: Option[String] = None,
+    componentGroup: Option[ComponentGroupName] = None,
+    disabled: Boolean = false
+) {
+
+  def toSingleComponentConfigWithoutValidators: SingleComponentConfig =
+    SingleComponentConfig(
+      params = Some(parameterConfigs.mapValuesNow(_.toParameterConfigWithoutValidators)),
+      icon = icon,
+      docsUrl = docsUrl,
+      componentGroup = componentGroup,
+      disabled = disabled,
+      componentId = None
+    )
+
+}
+
+case class ParameterAdditionalUIConfig(
+    required: Boolean = false,
+    initialValue: Option[FixedExpressionValue],
+    hintText: Option[String],
+    valueEditor: Option[ValueInputWithFixedValues],
+    valueCompileTimeValidation: Option[ParameterValueCompileTimeValidation],
+) {
+
+  def toParameterConfigWithoutValidators: ParameterConfig = ParameterConfig(
+    defaultValue = initialValue.map(_.expression),
```

Comment:
What about the whole `FragmentParameterValidator.validateFixedExpressionValues` stuff? I think we need to do some refactor and reuse this code. We shouldn't allow to break component definition using this API. Also, I figured out, that this code around `FixedValues` in fragments is in the wrong place. It is used in `extractValidatorsToParam` as a part of `FragmentCompleteDefinitionExtractor` which suppose to contain only validator's related stuff. The logic around other things suppose to be inside `FragmentWithoutValidatorsDefinitionExtractor` like the name describes.

Reply:
`validateFixedExpressionValues` required `expressionCompiler` and `validationContext`

not relying on them was the whole reason behind creating FragmentWithoutValidatorsDefinitionExtractor (though the aim was to ignore `valueCompileTimeValidation`, I didn't consider that it also doesn't validate the fixed values - this currently isn't a big problem as FragmentWithoutValidatorsDefinitionExtractor is used on already defined, validated beforehand, fragments -  but it should be made clearer)

if we require `AdditionalUIConfig.toSingleComponentConfig` to rely on `expressionCompiler` and `validationContext` it'd be difficult to use it in `prepareUIProcessObjects` - perhaps after refactoring the usage of AdditionalUIConfigProvider to happen at the level of ModelData or thereabouts it'd be easier

>We shouldn't allow to break component definition using this API

I agree, but this PR isn't a step back, with the old API (or just through config) one could provide nonsensical editors/validators for parameters and break them completely, here the worst that can happen is a `Fixed list` with no valid values.

Reply:
> validateFixedExpressionValues required expressionCompiler and validationContext
Ahh, I missed it. So it looks like we need to make some things clearer e.g. rename `extractValidatorsToParam` and extend comment next to `FragmentWithoutValidatorsDefinitionExtractor` that it also not validates `initialValue` and `valueEditor`.

> perhaps after refactoring the usage of AdditionalUIConfigProvider to happen at the level of ModelData or thereabouts it'd be easier
I agree. Let's add some TODO 

 > I agree, but this PR isn't a step back, with the old API ...
Yeah, the previous one already was too much low level. I thought that with a little effort and extraction of some code we can make things better. But I see know, that it isn't easy and also I was cheated by classes/methods names in the previous one PR which caused this wrong assumption. Please add some mentioned above clarifications / naming tweaks and we can move forward

Reply:
Now this code looks much better. It is easier to understand what are the responsibilities of each classes. Now, what do you think about reusing the `FragmentParameterValidator.validateFixedValuesList` also for non-fragment components? It doesn't need expression compiler.

Reply:
yep, i've reused it and a bit more, good idea